app/services/ui_notifications/hosts/build_completed.rb

```diff
@@ -0,0 +1,36 @@
+module UINotifications
+  module Hosts
+    class BuildCompleted < Base
+      private
+
+      def create
+        if blueprint.notifications.where(:subject => subject).empty?
+        add_notification
+        else
+          update_notifications
+        end
+      end
+
+      def add_notification
+        logger.debug('creating new UI notification host_build_completed')
+        ::Notification.create!(
+          initiator: initiator,
+          subject: subject,
+          audience: audience,
+          notification_blueprint: blueprint
+          )
+      end
+
+      def update_notifications
+        logger.debug('UI notification already exists, updating timestamps.')
+        blueprint.notifications.
+          where(subject: subject).
+          update_all(expired_at: blueprint.expired_at)
+      end
+
+      def blueprint
+        @blueprint ||= NotificationBlueprint.find_by(name: 'host_build_completed')
```

Comment:
We can cache this blueprint on class level - it will spare a query for each notification.